### ISSUE TITLE
Many-to-many associations using edges

### ISSUE THREAD

**Issue by dmarcelino** [appscot/waterline-orientdb#29]
Currently waterline-orientdb uses a junction table for many-to-many associations, just like SQL adapters. Even though this works it doesn't take advantage of using a graph db. Replace the junction table for edges.

Also waiting on balderdashy/waterline#725 in order to better name many-to-many edges.

This is not a priority for now as this can be circumvented by using Many-to-many **Through** associations which do use edges.

**Comment by dmarcelino** [comment]
Fixed by #46.
